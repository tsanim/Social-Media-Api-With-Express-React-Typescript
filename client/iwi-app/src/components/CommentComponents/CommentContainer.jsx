import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart, faTrash } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';

function CommentContainer(props) {
    return (
        <div className="commentContainer">
            {/* If edit form is shown , both comment's text and buttons are not shown */}
            <p>{props.text} </p>
            <div className="commentBtns">
                {
                    props.isLiked
                        ? <button onClick={props.handleDisLikeComment} className="liked"><FontAwesomeIcon icon={faHeart} /></button>
                        : <button onClick={props.handleLikeComment}><FontAwesomeIcon icon={faHeart} /></button>
                }

                {
                    (props.currentUser._id ? props.currentUser._id : props.currentUser.id) === props.commentCreator._id
                        ? <button onClick={props.handleShowModal} ><FontAwesomeIcon icon={faTrash} /></button>
                        : null
                }
            </div>
        </div>
    )
}

CommentContainer.propTypes = {
    text: PropTypes.string,
    isLiked: PropTypes.bool,
    currentUser: PropTypes.object,
    commentCreator: PropTypes.object,
    handleDisLikeComment: PropTypes.func,
    handleLikeComment: PropTypes.func,
    handleShowModal: PropTypes.func
}

export default CommentContainer;