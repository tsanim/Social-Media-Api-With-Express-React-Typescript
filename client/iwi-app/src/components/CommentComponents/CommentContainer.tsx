import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart, faTrash } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import CommentContainerProps from '../../interfaces/Components/CommentComponents/CommentContainerProps.interface';

function CommentContainer(props: CommentContainerProps) {
    return (
        <div className="commentContainer">
            {/* If edit form is shown , both comment's text and buttons are not shown */}
            <p>{props.text} </p>
            <div className="commentBtns">
                {
                    props.isLiked
                        ? <button onClick={props.handleDislikeComment as () => void} className="liked"><FontAwesomeIcon icon={faHeart} /></button>
                        : <button onClick={props.handleLikeComment as () => void}><FontAwesomeIcon icon={faHeart} /></button>
                }

                {
                    (props.currentUser._id ? props.currentUser._id : props.currentUser.id) === props.commentCreator._id
                        ? <button onClick={props.handleShowModal} ><FontAwesomeIcon icon={faTrash} /></button>
                        : null
                }
            </div>
        </div>
    )
}

CommentContainer.propTypes = {
    text: PropTypes.string,
    isLiked: PropTypes.bool,
    currentUser: PropTypes.object,
    commentCreator: PropTypes.object,
    handleDisLikeComment: PropTypes.func,
    handleLikeComment: PropTypes.func,
    handleShowModal: PropTypes.func
}

export default CommentContainer;