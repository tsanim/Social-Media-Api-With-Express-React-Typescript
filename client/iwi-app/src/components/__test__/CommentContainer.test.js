import React from 'react';
import CommentContainer from '../CommentComponents/CommentContainer';

import { shallow } from 'enzyme';

function commentContainerComponentSetup() {
    const props = {
        text: '',
        isLiked: true,
        currentUser: { _id: '123' },
        commentCreator: { _id: '123' },
        handleDisLikeComment: jest.fn(),
        handleLikeComment: jest.fn(),
        handleShowModal: jest.fn()
    }

    const enzymeWrapper = shallow(<CommentContainer {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('CommentContainer component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = commentContainerComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div tag with class "commentContainer"', () => {
        const { enzymeWrapper } = commentContainerComponentSetup();


        expect(enzymeWrapper.find('div.commentContainer')).toHaveLength(1);
    });

    it('should render div tag with class "commentBtns"', () => {
        const { enzymeWrapper } = commentContainerComponentSetup();

        expect(enzymeWrapper.find('div.commentBtns')).toHaveLength(1);
    });

    it('should render div tag with class "commentBtns"', () => {
        const { enzymeWrapper } = commentContainerComponentSetup();

        expect(enzymeWrapper.find('div.commentBtns')).toHaveLength(1);
    });

    it('should render 3 button tags if isLiked is true and currentUser id is equal to commentCreator id', () => {
        const { enzymeWrapper } = commentContainerComponentSetup();

        expect(enzymeWrapper.find('button')).toHaveLength(2);
    });
});