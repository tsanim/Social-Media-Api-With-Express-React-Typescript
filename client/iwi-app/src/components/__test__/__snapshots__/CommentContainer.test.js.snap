// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer component should render initial layout 1`] = `
Array [
  <div
    className="commentContainer"
  >
    <p>
      
       
    </p>
    <div
      className="commentBtns"
    >
      <button
        className="liked"
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f004",
                "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
              ],
              "iconName": "heart",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <button
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f1f8",
                "M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z",
              ],
              "iconName": "trash",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </div>,
]
`;
